src/index.ts

```diff
@@ -172,10 +172,57 @@ app.get("/me", async (c) => {
     login.account.id,
   );
 
-  return c.json({
-    name,
-    email: login.account.email,
-  });
+  //roleを確認
+  const roleEntry = await c.env.DB
+  .prepare("SELECT role FROM account WHERE id = ?")
+  .bind(login.account.id)
+  .first();
+  if ( roleEntry === null ){
+    throw new Error("Entry was invalid");
+  }
+
+  const { role } = roleEntry;
+  //学生であった場合registrationから科目idを取り出す
+  if ( role === 'STUDENT'){
+    const subjectEntry = c.env.DB
+    .prepare("SELECT subject_id FROM registration WHERE sutudent_id = ?")
+    .bind(login.account.id)
+    .raw();
+
+    if ( subjectEntry === null ){
+      throw new Error("Entry was invalid");
+    }
+
+    const subject = subjectEntry;
+    
+    return c.json({
+      name,
+      email: login.account.email,
+      registrations: subject,
+    });
+  }
+  //教師であった場合chargeから科目idを取り出す
+  else if ( role === 'TEACHER' ){
+    const subjectEntry = await c.env.DB
+    .prepare("SELECT subject_id FROM charge WHERE teacher_id = ?")
+    .bind(login.account.id)
+    .raw();
+
+    if ( subjectEntry === null ){
+      throw new Error("Entry was invalid");
+    }
+
+    const subject = subjectEntry;
+    
+    return c.json({
+      name,
+      email: login.account.email,
+      charges: subject,
+    });
+  }
+  else {
+    throw new Error("Role was invalid");
+  }
```

Comment:
```suggestion
  throw new Error(`unknown role: ${role}`);
```
この `else` は不要です. また, 知らないロールであったことを表すエラーのほうがよいかと思います.